### PR TITLE
Fixing extra vacols calls

### PR DESCRIPTION
**Pull request by mdbenjam**
### Description
The frontend refactor started making a lot of VACOLS requests. This is because the legacy_task does not memoize its relationship with appeal. And while we grab appeals in bulk we need to assign them to the task model preemptively.

### Acceptance Criteria 
- [ ] Should not see any `load_vacols_data` calls when loading a Queue.

### CODE REVIEW

File: app/models/queues/legacy_work_queue.rb [hunk 1]
```diff
@@ -37,8 +37,11 @@ def tasks_with_appeals_of_vacols_tasks(user, role, vacols_tasks)
       vacols_appeals = repository.appeals_from_tasks(vacols_tasks)
 
       tasks = vacols_tasks.zip(vacols_appeals).map do |task, appeal|
-        MODEL_CLASS_OF_ROLE[role.capitalize].from_vacols(task, appeal, user)
+        return_task = MODEL_CLASS_OF_ROLE[role.capitalize].from_vacols(task, appeal, user)
```
Comment (aroltsch) [inline app/models/queues/legacy_work_queue.rb]: Should we set the appeal in from_vacols method?

Comment (mdbenjam) [reply app/models/queues/legacy_work_queue.rb]: I'd like to leave it here for now since it's only one place instead of two?

Comment (aroltsch) [reply app/models/queues/legacy_work_queue.rb]: I was suggesting to move it there

Comment (mdbenjam) [reply app/models/queues/legacy_work_queue.rb]: Oh I get in the base class's from_vacols. Sure that makes sense 👍

File: app/models/legacy_tasks/legacy_task.rb [hunk 2]
```diff
@@ -44,7 +45,8 @@ def self.from_vacols(record, appeal, user)
       assigned_to: user,
       task_id: record.created_at ? record.vacols_id + "-" + record.created_at.strftime("%Y-%m-%d") : nil,
       document_id: record.document_id,
-      assigned_by: record.assigned_by
+      assigned_by: record.assigned_by,
+      appeal: appeal
```
Comment (aroltsch) [inline app/models/legacy_tasks/legacy_task.rb]: Nice!

**Pull request merged**